actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -414,6 +414,20 @@ func (pb *Actor) Cancel(vmctx exec.VMContext, chid *types.ChannelID) (uint8, err
 			return errors.NewFaultError("Expected PaymentChannel from channels lookup")
 		}
 
+		if channel.Redeemed {
+			if channel.Condition == nil {
+				return errors.NewFaultError("Channel cannot be cancelled due to successful redeem")
+			} else {
+				err := checkCondition(vmctx, channel, channel.Condition, []interface{}{})
+				if err != nil && !errors.IsFault(err) {
```

Comment:
BLOCKING: If the error is not a revert error (`IsRevert`) it should be returned. Faults (and bad programmer) errors should be returned immediately.

Reply:
Specifically
1. Check err == nil. if so return coded revert error.
2. Check if `!errors.IsRevert(err)`. If so return error unmodified.,
3. else continue.

Reply:
`IsRevert` isn't a method implemented anywhere. Do you want me to write it?

Reply:
Do you mean `ShouldRevert`?

Reply:
(Answered offline: Yes)